src/libstd/sys/unix/thread.rs

```diff
@@ -52,6 +52,11 @@ impl Thread {
         assert_eq!(libc::pthread_attr_init(&mut attr), 0);
 
         let stack_size = cmp::max(stack, min_stack_size(&attr));
+
+        // L4Re only supports a maximum of 1Mb per default.
+        #[cfg(target_os = "l4re")]
+        let stack_size = cmp::min(stack_size, 1024 * 1024);
```

Comment:
Shouldn't we rely on `pthread_create` to return an error for larger stack sizes here?

Reply:
@humenda would it be possible to change the `min_stack_size` function instead of adding this clause here?